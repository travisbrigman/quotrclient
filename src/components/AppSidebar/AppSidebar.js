import { Box, Button, Collapsible, Layer } from "grommet";
import { FormClose } from "grommet-icons";

export default function AppSidebar({
  children,
  onClose,
  showSidebar,
  size,
  ...props
}) {
  if (!showSidebar || size !== "small") {
    return (
      <Collapsible direction="horizontal" open={showSidebar}>
        <Box
          flex
          width="medium"
          background="light-2"
          elevation="small"
          align="center"
          justify="center"
        >
          {children}
        </Box>
      </Collapsible>
    );
  }

  return (
    <Layer>
      <Box
        background="light-2"
        tag="header"
        justify="end"
        align="center"
        direction="row"
      >
        <Button
          a11yTitle="close sidebar"
          icon={<FormClose />}
          onClick={onClose}
        />
      </Box>
      <Box fill background="light-2" align="center" justify="center">
        {children}
      </Box>
    </Layer>
  );
}
