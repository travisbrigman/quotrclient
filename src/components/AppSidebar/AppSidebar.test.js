import AppSidebar from "./AppSidebar";
import React from "react";
import { render } from "@testing-library/react";

test("can render", () => {
  const { container } = render(
    <AppSidebar>
      <p>Sidebar</p>
    </AppSidebar>
  );

  expect(container).toMatchSnapshot();
});

test("can show", () => {
  const { container } = render(
    <AppSidebar showSidebar={true}>
      <p>Sidebar shown</p>
    </AppSidebar>
  );

  expect(container).toMatchSnapshot();
});

test("can hide", () => {
  const { container } = render(
    <AppSidebar showSidebar={false}>
      <p>Sidebar not shown</p>
    </AppSidebar>
  );

  expect(container).toMatchSnapshot();
});

test("can show small", () => {
  const { baseElement } = render(
    <AppSidebar size="small" showSidebar={true}>
      <p>Small sidebar shown</p>
    </AppSidebar>
  );

  expect(baseElement).toMatchSnapshot();
});

test("can show large", () => {
  const { container } = render(
    <AppSidebar size="large" showSidebar={true}>
      <p>Large sidebar shown</p>
    </AppSidebar>
  );

  expect(container).toMatchSnapshot();
});
