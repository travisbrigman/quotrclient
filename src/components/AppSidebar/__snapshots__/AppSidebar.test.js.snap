// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can hide 1`] = `
<div>
  <div
    aria-hidden="true"
    class="StyledBox-sc-13pk1d4-0 dYebPD Collapsible__AnimatedBox-sc-15kniua-0 hSCcJL"
  />
</div>
`;

exports[`can render 1`] = `
<div>
  <div
    aria-hidden="true"
    class="StyledBox-sc-13pk1d4-0 dYebPD Collapsible__AnimatedBox-sc-15kniua-0 hSCcJL"
  />
</div>
`;

exports[`can show 1`] = `
<div>
  <div
    aria-hidden="false"
    class="StyledBox-sc-13pk1d4-0 dYebPD Collapsible__AnimatedBox-sc-15kniua-0 gmfzxr"
    open=""
  >
    <div
      class="StyledBox-sc-13pk1d4-0 fbVFwI"
    >
      <p>
        Sidebar shown
      </p>
    </div>
  </div>
</div>
`;

exports[`can show large 1`] = `
<div>
  <div
    aria-hidden="false"
    class="StyledBox-sc-13pk1d4-0 dYebPD Collapsible__AnimatedBox-sc-15kniua-0 gmfzxr"
    open=""
  >
    <div
      class="StyledBox-sc-13pk1d4-0 fbVFwI"
    >
      <p>
        Large sidebar shown
      </p>
    </div>
  </div>
</div>
`;

exports[`can show small 1`] = `
<body>
  <div />
  <div>
    <div
      aria-hidden="false"
    >
      <div
        class="StyledLayer-rmtehz-0 gNIvtm"
        tabindex="-1"
      >
        <div
          class="StyledLayer__StyledOverlay-rmtehz-1 iconax"
        />
        <div
          class="StyledLayer__StyledContainer-rmtehz-2 icgHnb"
          data-g-portal-id="0"
        >
          <a
            aria-hidden="true"
            class="LayerContainer__HiddenAnchor-sc-1srj14c-0 chPKWK"
            tabindex="-1"
          />
          <header
            class="StyledBox-sc-13pk1d4-0 bUnQxS"
          >
            <button
              aria-label="close sidebar"
              class="StyledButton-sc-323bzc-0 hiyJPE"
              type="button"
            >
              <svg
                aria-label="FormClose"
                class="StyledIcon-ofa7kd-0 dEeaTZ"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7,7 L17,17 M7,17 L17,7"
                  fill="none"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </button>
          </header>
          <div
            class="StyledBox-sc-13pk1d4-0 kkVWtn"
          >
            <p>
              Small sidebar shown
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
